Google Scholar.js

```diff
@@ -661,6 +570,630 @@ ItemFactory.prototype.saveItemCommonVars = function () {
+	// for the BibTeX document.
+	await delay(DELAY_INTERVAL);
+
+	// NOTE: To emulate the web app, the referrer for the BibTeX text is always
+	// set to the origin (e.g. https://scholar.google.com/), imitating
+	// strict-origin-when-cross-origin
+	reqOptions.headers.Referer = GS_CONFIG.baseURL + "/";
+	const bibTeXBody = await ZU.requestText(bibTeXURL, reqOptions);
+
+	let translator = Z.loadTranslator("import");
+	translator.setTranslator("9cb70025-a888-4a29-a210-93ec52da40d4"); // BibTeX
+	translator.setString(bibTeXBody);
+	translator.setHandler("itemDone", function (obj, item) {
+		// case are not recognized and can be characterized by the
+		// title link, or that the second line starts with a number
+		// e.g. 1 Cr. 137 - Supreme Court, 1803
+		if ((row.directLink && row.directLink.includes('/scholar_case?'))
+			|| row.byline && "01234567890".includes(row.byline[0])) {
+			item.itemType = "case";
+			item.caseName = item.title;
+			item.reporter = item.publicationTitle;
+			item.reporterVolume = item.volume;
+			item.dateDecided = item.date;
+			item.court = item.publisher;
+		}
+		// patents are not recognized but are easily detected
+		// by the titleLink or second line
+		if ((row.directLink && row.directLink.includes('google.com/patents/'))
+			|| (row.byline && row.byline.includes('Google Patents'))) {
+			item.itemType = "patent";
+			// authors are inventors
+			for (let i = 0, n = item.creators.length; i < n; i++) {
+				item.creators[i].creatorType = 'inventor';
+			}
+			// country and patent number
+			if (row.directLink) {
+				let m = row.directLink.match(/\/patents\/([A-Za-z]+)(.*)$/);
+				if (m) {
+					item.country = m[1];
+					item.patentNumber = m[2];
+				}
+			}
+		}
+
+		// Add the title link as the url of the item.
+		if (row.directLink) {
+			item.url = row.directLink;
+		}
+
+		// fix titles in all upper case, e.g. some patents in search results
+		if (item.title.toUpperCase() === item.title) {
+			item.title = ZU.capitalizeTitle(item.title);
+		}
+
+		// delete "others" as author
+		if (item.creators.length) {
+			var lastCreatorIndex = item.creators.length - 1,
+				lastCreator = item.creators[lastCreatorIndex];
+			if (lastCreator.lastName === "others" && (lastCreator.fieldMode === 1 || lastCreator.firstName === "")) {
+				item.creators.splice(lastCreatorIndex, 1);
+			}
+		}
+
+		// clean author names
+		for (let j = 0, m = item.creators.length; j < m; j++) {
+			if (!item.creators[j].firstName) {
+				continue;
+			}
+
+			item.creators[j] = ZU.cleanAuthor(
+				item.creators[j].lastName + ', '
+					+ item.creators[j].firstName,
+				item.creators[j].creatorType,
+				true);
+		}
+
+		addAttachment(item, row);
+
+		item.complete();
+	});
+	return translator.translate();
+}
+
+function addAttachment(item, row) {
+	// attach linked document as attachment if available
+	if (row.attachmentLink) {
+		let	attachment = {
+			title: "Full Text (via Google Scholar)",
```

Comment:
Hmm, yeah. This is tough.

I think "Linked Document" is too vague. Linked from where?

For files from Unpaywall, we use "Submitted Version", "Accepted Version", and "Published Version", but we don't know what those are here. Could be "Google Scholar Linked Version", but that's not really appropriate for the same reason "Full Text (via Google Scholar)" isn't really appropriate — the fact that it came from Google Scholar isn't really relevant to what it is. "Available Version"? "Publicly Available Version"?

Reply:
FWIW, I actually liked the (via Google Scholar) option because it allows quick searching/filtering by the attachment title and also makes clear to users why they get a 'wrong' (e.g., old working paper) PDF. Otoh, with "Available Version" it's unclear where it's available from and, where it's wrong, it looks like Zotero mishandled it.

Reply:
That's fair. I guess the practical aspect of this outweighs any conceptual objections to putting that in the title. OK, @zoe-translates, so we can revert to that.

Reply:
OK, sorry, let's go with `Available Version (Google Scholar)`. `Available Version via Google Scholar` is vaguely ungrammatical, but `Available Version` is nice and consistent, and `(Google Scholar)` makes sense as a sort of tag. Put them together and we have a winner.